update documentation of select (#2011)
JuliaData/DataFrames.jl: docs/src/man/getting_started.md

true
```

> In the first cases, `[:A]` is a vector, indicating that the resulting object should be a `DataFrame`, since a vector can contain one or more column names. On the other hand, `:A` is a single symbol, indicating that a single column vector should be extracted. Note, that in the first case a vector is required to be passed (not just any iterable), so e.g. `df[:, (:x1, :x2)]` is not allowed, but `df[:, [:x1, :x2]]` is valid.

Reviewer: ```suggestion
In the first cases, `[:A]` is a vector, indicating that the resulting object should be a `DataFrame`, since a vector can contain one or more column names. On the other hand, `:A` is a single symbol, indicating that a single column vector should be extracted. Note that in the first case a vector is required to be passed (not just any iterable), so e.g. `df[:, (:x1, :x2)]` is not allowed, but `df[:, [:x1, :x2]]` is valid.
```

In the first cases, `[:A]` is a vector, indicating that the resulting object should be a `DataFrame`, since a vector can contain one or more column names. On the other hand, `:A` is a single symbol, indicating that a single column vector should be extracted.

> Note, that similarly to Base Julia you are not allowed to select rows nor columns using `Tuple`s, use `AbstractVector`s instead.

Author: I have shortened it (I decided to add it as people ask about it - so it seems it creates confusion for newcomers).

For me the general answer to all the questions like this would be "it works like Base Julia" period (this was the design goal we had ~1 year ago and now we have it) 😄.

```

> You can also use the [`select`](@ref) and [`select!`](@ref) functions to select columns in a data frame.

Author: OK - I have moved it down to a separate subheading (so now we have two subheadings - one for indexing and one for selecting)

Equivalently, the `in` function can be called with a single argument to create a function object that tests whether each value belongs to the subset (partial application of `in`): `df[in([1, 5, 601]).(df.A), :]`.

> #### Column selection syntax

Reviewer: ```suggestion
#### Column selection using `select` and `select!`
```

> You can also use the [`select`](@ref) and [`select!`](@ref) functions to select columns in a data frame.

Reviewer: How about mentioning the main differences between `select` and indexing, namely: 1) that `select(df, :x1)` returns a data frame rather than a vector? Also, `select` makes copies like `df[:, cols]`, which isn't obvious.

Author: added

The `select` function creates a new data frame:
```jldoctest dataframe
julia> df = DataFrame(x1=1, x2=2, y=3)
1×3 DataFrame
│ Row │ x1    │ x2    │ y     │
│     │ Int64 │ Int64 │ Int64 │
├─────┼───────┼───────┼───────┤
│ 1   │ 1     │ 2     │ 3     │

julia> select(df, Not(:x1)) # drop column :x1 in a new data frame
1×2 DataFrame
│ Row │ x2    │ y     │
│     │ Int64 │ Int64 │
├─────┼───────┼───────┤
│ 1   │ 2     │ 3     │

julia> select(df, r"x") # select columns containing 'x' character
1×2 DataFrame
│ Row │ x1    │ x2    │
│     │ Int64 │ Int64 │
├─────┼───────┼───────┤
│ 1   │ 1     │ 2     │
```

> To perform the operations in-place use `select!`:

Reviewer: ```suggestion

To perform the operations in-place use `select!`:
```

#### Column selection using `select` and `select!`

You can also use the [`select`](@ref) and [`select!`](@ref) functions to select columns in a data frame.

The `select` function creates a new data frame:
```jldoctest dataframe
julia> df = DataFrame(x1=1, x2=2, y=3)
1×3 DataFrame
│ Row │ x1    │ x2    │ y     │
│     │ Int64 │ Int64 │ Int64 │
├─────┼───────┼───────┼───────┤
│ 1   │ 1     │ 2     │ 3     │

julia> select(df, Not(:x1)) # drop column :x1 in a new data frame
1×2 DataFrame
│ Row │ x2    │ y     │
│     │ Int64 │ Int64 │
├─────┼───────┼───────┤
│ 1   │ 2     │ 3     │

julia> select(df, r"x") # select columns containing 'x' character
1×2 DataFrame
│ Row │ x1    │ x2    │
│     │ Int64 │ Int64 │
├─────┼───────┼───────┤
│ 1   │ 1     │ 2     │
```

It is important to note that `select` always returns a data frame,
even if a single column is selected (as opposed to indexing syntax).
```jldoctest dataframe
julia> select(df, :x1)
1×1 DataFrame
│ Row │ x1    │
│     │ Int64 │
├─────┼───────┤
│ 1   │ 1     │

julia> df[:, :x1]
1-element Array{Int64,1}:
 1
```

By default `select` copies columns of a passed source data frame.

> In order to avoid copying set its `copycols` keyword argument to `false`:

Reviewer: ```suggestion
In order to avoid copying, pass `copycols=false`:
```

```
julia> df2 = select(df, :x1)
1×1 DataFrame
│ Row │ x1    │
│     │ Int64 │
├─────┼───────┤
│ 1   │ 1     │

julia> df2.x1 === df.x1
false

julia> df2 = select(df, :x1, copycols=false)
1×1 DataFrame
│ Row │ x1    │
│     │ Int64 │
├─────┼───────┤
│ 1   │ 1     │

julia> df2.x1 === df.x1
true
```

To perform the selection operation in-place use `select!`:
```jldoctest dataframe

> julia> select!(df, Not(:x1))

Reviewer: I'd show `df` right after this, instead of the result of `select!`. Then you can drop the second example.

Author: fixed - is this what you had in mind?

```

> If you are interested in describing only a subset of columns then the easiest way to do it is to

Author: Right - fixed.

Regarding an additional arg do `describe` I have opened https://github.com/JuliaData/DataFrames.jl/issues/2014 to keep it in mind (in general I agree we could add it).
